src/frontend/src/common/queries/trees.ts

```diff
@@ -200,3 +202,58 @@ export function useDeleteTree({
     },
   });
 }
+
+/**
+ * edit trees
+ */
+
+interface EditTreePayloadType {
+  name: string;
+}
+
+interface TreeEditRequestType {
+  treeIdToEdit: string;
+  newTreeName: string;
+}
+
+interface TreeEditResponseType {
+  id: string;
+}
+
+type TreeEditCallbacks = MutationCallbacks<TreeEditResponseType>;
+
+export async function editTree({
+  treeIdToEdit,
+  newTreeName,
+}: TreeEditRequestType): Promise<TreeEditResponseType> {
+  const payload: EditTreePayloadType = {
+    name: newTreeName,
+  };
+  const response = await fetch(API_URL + API.PHYLO_TREES_V2 + treeIdToEdit, {
```

Comment:
If you have the time, I think this would be a great candidate to make a generic `putBackendApiJson` ala my previous `getBackendApiJson`, then just use it here. You don't have to convert over anything else right now, but I want to standardize our request functions so we ensure consistency in our API requests and don't have so many bare `fetch` calls in the code.

Reply:
i'll do this when i add the update samples request, I sort of just want to get this PR merged in as soon as possible now 😓